components/heading/src/index.js

```diff
@@ -0,0 +1,120 @@
+import styled from 'styled-components';
+import React, { createElement } from 'react';
+import PropTypes from 'prop-types';
+import {
+  HEADING_SIZES,
+  LEVEL_SIZE,
+  LEVEL_TAG,
+  MEDIA_QUERIES,
+  TYPOGRAPHY_SCALE,
+} from '@govuk-react/constants';
+import { spacing, typography } from '@govuk-react/lib';
+
+// use `size` only with string for XLARGE, SMALL etc and number for px size
+// so if `size` is a string, we find a numeric size based off `HEADING_SIZES`
+// but if `size` is a number we just send through that number
+
+const StyledHeading = styled(({
+  level, children, size, ...props
+}) =>
+  createElement(LEVEL_TAG[level], props, children))(
```

Comment:
this is old code but looking at it now I guess it stops us using the `as` prop with `Heading`.

Should we raise an issue for this?

Reply:
just tried it out as i thought it wouldn't affect things...

When using `<H1 as="h2">` you lose all styles - not good

yes - we should raise a separate issue